### PR TITLE
Risk analysis (Milestone 11)

### PR DESCRIPTION
**Pull request by z-arnott**
Add risk analysis for package, vulnerability and system (Milestone 11)
Note that Task 11.2, Develop Analysis Controller component, will come in a later PR, once this component is integrated with storage facade.

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -1,4 +1,6 @@
+import { Package, SbomFormat } from '../src/utils/types.utils';
 import { parse } from '../src/services/parserContext.services';
+import { AnalysisInfo } from '../src/services/riskAnalysis.services';
```
Comment (Seneli) [inline src/index.ts:3]: looks like `AnalysisInfo` and `Package, SbomFormat` aren't used even though they are imported. We can remove redundant imports :)

File: src/index.ts [hunk 2]
```diff
@@ -25,6 +27,9 @@ app.post('/upload', (req: any, res: any, next: any) => {
   //add middleware calls here as needed
 });
 
+app.get('/riskanalysis', (req: any, res: any, next: any) => {
```
Comment (Seneli) [inline src/index.ts:30]: im a bit unsure but if we have a route handler via `next` should we have a case where we call it? or is this gonna be integrated in the next PR with **11.2 Develop Analysis Controller component?**

File: src/services/riskAnalysis.services.ts [hunk 3]
```diff
@@ -1,289 +1,328 @@
...
+ * Object representing the results of a system analysis
+ */
+interface AnalysisInfo {
+  sessionID: number;
+  packagesAnalyzed: number;
+  systemRisk: number;
+}
+
+/**
+ * Find the risk of one a software system based on the risk of the system's packages
```
Comment (Seneli) [inline src/services/riskAnalysis.services.ts:18]: typo: should be "of one software system..."

**Review by Seneli — APPROVED** [review]
Looks good just some minor changes that I noted above. Full disclaimer, for the risk analysis functions I mostly just read through the code. I'll take your word for the CVSS 3->2 functions and so on. It's very easy to understand though, so it should be easy to modify in the future if we ever need to :) 

Also, I think I need to set up my supabase key in the env file before the tests can run properly. I'll get that from y'all. Since that's not a part of this PR I'll approve tho. You're killing it as always!
Screen Shot 2023-01-19 at 3 32 20 PM

File: src/index.ts (hunk 2 above)
Comment (z-arnott) [reply src/index.ts:30]: Implemented in the next PR, I was waiting for the storage facade before I could proceed

**Pull request merged**